pkg/download/blob_test.go

```diff
@@ -145,14 +151,65 @@ func Test_blobDownload_actualBlob(t *testing.T) {
 		Blob:        name,
 		StorageBase: base,
 	})
-	_, body, err := Download(d)
+	_, body, err := Download(testctx, d)
 	require.Nil(t, err)
 	defer body.Close()
 	b, err := ioutil.ReadAll(body)
 	require.Nil(t, err)
 	require.EqualValues(t, chunk, b, "retrieved body is different body=%d chunk=%d", len(b), len(chunk))
 }
 
+func Test_blobDownload_fails_actualBlob404(t *testing.T) {
+	acct := os.Getenv("AZURE_STORAGE_ACCOUNT")
+	key := os.Getenv("AZURE_STORAGE_ACCESS_KEY")
+	if acct == "" || key == "" {
+		t.Skipf("Skipping: AZURE_STORAGE_ACCOUNT or AZURE_STORAGE_ACCESS_KEY not specified to run this test")
+	}
+	base := storage.DefaultBaseURL
+
+	blobName := "<BLOB THAT DOESN'T EXIST>"
+	containerName := "<CONTAINER NAME>"
+
+	// Get the blob via downloader
+	d := NewBlobDownload(acct, key, blobutil.AzureBlobRef{
+		Container:   containerName,
+		Blob:        blobName,
+		StorageBase: base,
+	})
+	code, _, err := Download(testctx, d)
+	require.NotNil(t, err)
+	require.Equal(t, code, http.StatusNotFound)
+	require.Contains(t, err.Error(), "because it does not exist")
+	require.Contains(t, err.Error(), "Not Found")
+	require.Contains(t, err.Error(), "Service request ID")
+}
+
+func Test_blobDownload_fails_actualBlob409(t *testing.T) {
+	// before running this test, go to your storage account on portal > Configuration and disable Blob public access
```

Comment:
It's great that you tested with actual blobs, but the unit tests will need to mock these failures or they'll fail in the lab and on others' machines